internal/config/config.go

```diff
@@ -323,198 +329,40 @@ func Default() *Config {
 	}
 }
 
-func validateK8sVersion(fl validator.FieldLevel) bool {
-	return versions.IsSupportedK8sVersion(fl.Field().String())
-}
-
-func validateAWSInstanceType(fl validator.FieldLevel) bool {
-	return validInstanceTypeForProvider(fl.Field().String(), false, cloudprovider.AWS)
-}
-
-func validateAzureInstanceType(fl validator.FieldLevel) bool {
-	azureConfig := fl.Parent().Interface().(AzureConfig)
-	var acceptNonCVM bool
-	if azureConfig.ConfidentialVM != nil {
-		// This is the inverse of the config value (acceptNonCVMs is true if confidentialVM is false).
-		// We could make the validator the other way around, but this should be an explicit bypass rather than checking if CVMs are "allowed".
-		acceptNonCVM = !*azureConfig.ConfidentialVM
-	}
-	return validInstanceTypeForProvider(fl.Field().String(), acceptNonCVM, cloudprovider.Azure)
-}
-
-func validateGCPInstanceType(fl validator.FieldLevel) bool {
-	return validInstanceTypeForProvider(fl.Field().String(), false, cloudprovider.GCP)
-}
-
-// validateProvider checks if zero or more than one providers are defined in the config.
-func validateProvider(sl validator.StructLevel) {
-	provider := sl.Current().Interface().(ProviderConfig)
-	providerCount := 0
-
-	if provider.AWS != nil {
-		providerCount++
-	}
-	if provider.Azure != nil {
-		providerCount++
-	}
-	if provider.GCP != nil {
-		providerCount++
-	}
-	if provider.QEMU != nil {
-		providerCount++
-	}
-
-	if providerCount < 1 {
-		sl.ReportError(provider, "Provider", "Provider", "no_provider", "")
-	} else if providerCount > 1 {
-		sl.ReportError(provider, "Provider", "Provider", "more_than_one_provider", "")
+// FromFile returns config file with `name` read from `fileHandler` by parsing
+// it as YAML.
+func FromFile(fileHandler file.Handler, name string) (*Config, error) {
```

Comment:
optional: We can try to prevent developers from accidentially using this func instead of New. I've got two suggestions:
* This func is currently only used by cdbg. If it's acceptable that cdbg validates the config, we could use New there, too, and unexport this func.
* Or we could extend the doc comment to encourage developers to use New in most cases.